src/primitives/transaction.cpp

```diff
@@ -57,8 +57,10 @@ JSDescription JSDescription::Randomized(
     // Randomize the order of the inputs and outputs
     inputMap = {0, 1};
     outputMap = {0, 1};
-    MappedShuffle(inputs.begin(), inputMap.begin(), ZC_NUM_JS_INPUTS, gen);
-    MappedShuffle(outputs.begin(), outputMap.begin(), ZC_NUM_JS_OUTPUTS, gen);
+    if (gen) {
```

Comment:
Also, without this conditional section, if a user passes in an empty func, MappedShuffle is still invoked and the behaviour is undefined and an exception is thrown.